### ISSUE TITLE
Enable for all markdown files in a PR

### ISSUE THREAD

**Issue by RichardLitt** [dkhamsing/awesome_bot#77]
Would be great to not limit this to README.md, but any markdown files in a PR.

**Comment by dkhamsing** [comment]
@RichardLitt uh you could support multiple files now, just add another line in the ci config (.travis.yml)

```yml
#...
before_script:
  - gem install awesome_bot
script:
  - awesome_bot README.md 
  - awesome_bot plans.md
# etc   
```

**Comment by RichardLitt** [comment]
Does that work with `*.md`?

**Comment by dkhamsing** [comment]
no

**Comment by RichardLitt** [comment]
Hmm. I should have been more clear, then. It would be useful to enable all markdown files, not just explicitly named ones, to be checked. 

I'll see if I can PR this.

**Comment by dkhamsing** [comment]
Ah
Actually check this out https://github.com/dkhamsing/awesome_bot/pull/76

**Comment by RichardLitt** [comment]
Ah, sweet! Ok, this can be closed when #76 is merged. 

I just spent 30~ trying to get this working, and I've come to the conclusion I don't know Ruby enough. I'm not sure how to... lint a file? How do you debug?

Given that, I'm going to wait on #76. Seems smarter. :)

**Comment by dkhamsing** [comment]
no worries

**Comment by sotayamashita** [comment]
@RichardLitt you can use my fork version. Here is how to use it:

**Comment by RichardLitt** [comment]
@sotayamashita Sweet. I'll wait until this is merged - I use `awesome_bot` in a few places, and don't want to change them all quite yet. I'm not in the largest rush.

**Comment by dkhamsing** [comment]
will work with @sotayamashita to get his work merged
thanks!

**Comment by dkhamsing** [comment]
@RichardLitt pushed new version `1.9.0` that supports `*.md`
Let me know if you have questions

**Comment by RichardLitt** [comment]
@dkhamsing Awesome! Thanks so much for this. 

Does this work for only files which have been pushed to a branch (as in a PR?)

**Comment by dkhamsing** [comment]
Uh... yes?

Here is an example https://github.com/ReadmeCritic/open-source-android-apps/blob/patch-1/.travis.yml

What use case are you trying to support?

**Comment by RichardLitt** [comment]
Hmm. I can't see that link (or repo), not sure why. 

I mean something like this: https://github.com/ipfs/newsletter/pull/44/files. The check ran for README.md. What I would like is for it to run only for the markdown file in the PR. Do you know how I could do that?

**Comment by dkhamsing** [comment]
The link was deleted, here's the correct one https://github.com/pcqpcq/open-source-android-apps/blob/master/.travis.yml#L10

**Comment by dkhamsing** [comment]
@RichardLitt hm I see, you could try awesome_bot published/*.md but it would check old files as well, not just the new one

**Comment by RichardLitt** [comment]
Hmm. Cool. I can also do `**/*.md` to check everything - works with the CLI tool (which is awesome). 

I'm not sure what would be needed to only check staged files, or files in a given commit.

**Comment by RichardLitt** [comment]
I'm not sure this issue should be closed yet, as the original intent hasn't been met.

**Comment by dkhamsing** [comment]
hm getting the diff in a pull request is out of scope for the tool but it can probably be done with `git diff`

**Comment by dkhamsing** [comment]
Will post an update if I figure this out

**Comment by RichardLitt** [comment]
Working on it here: https://travis-ci.org/ipfs/newsletter/builds/130925178

```
git diff master.. --name-only | grep '.md' | xargs awesome_bot --white-list https://github.com/ipfs/newsletter/issues/new
```

**Comment by dkhamsing** [comment]
nice!

**Comment by RichardLitt** [comment]
Ok, yep: this totally works! All good. Thanks so much, @dkhamsing and @sotayamashita. :)
